packages/frontend/src/Build.svelte

```diff
@@ -67,53 +66,80 @@ async function fillBuildOptions() {
+  let prereqs = await bootcClient.checkPrereqs();
+  if (prereqs) {
+    errorFormValidation = prereqs;
+    existingBuild = false;
+    return;
+  }
+
+  if (!selectedImage) {
+    errorFormValidation = 'No image selected';
+    existingBuild = false;
+    return;
+  }
+
+  if (!buildFolder) {
+    errorFormValidation = 'No output folder selected';
+    existingBuild = false;
+    return;
+  }
+
+  if (!buildType || buildType.length === 0) {
+    errorFormValidation = 'Must select at least one disk image type';
+    existingBuild = false;
+    return;
+  }
+
+  if (!buildArch) {
+    errorFormValidation = 'At least one architecture must be selected';
+    existingBuild = false;
+    return;
+  }
+
+  // overwrite
+  existingBuild = await bootcClient.buildExists(buildFolder, buildType);
+  if (existingBuild && !overwrite) {
+    errorFormValidation = 'Confirm overwriting existing build';
+    return;
+  }
+
+  // no problems, ready to build!
+  errorFormValidation = undefined;
+}
 
-  // Fields to check and it's output
-  const fieldsToCheck = [
-    { key: 'image name', value: buildImageName },
-    { key: 'tag', value: buildTag },
-    { key: 'engine ID', value: buildEngineId },
-    { key: 'build folder', value: buildFolder },
-    { key: 'build type', value: buildType },
-    { key: 'architecture', value: buildArch },
-  ];
-
-  // Go through and check if any of the fields are missing
-  // and create an error message if so
-  let missingFields: string[] = [];
-  fieldsToCheck.forEach(field => {
-    if (!field.value) missingFields.push(field.key);
-  });
-  if (missingFields.length > 0) {
-    errorFormValidation = `Missing fields: ${missingFields.join(', ')}`;
+async function buildBootcImage() {
+  // you can't get here without a selected image, but this
+  // avoids a svelte error
+  if (!selectedImage) {
     return;
   }
 
   // Before building a disk image name, we get a unique unused identifier for this image
   // This is to prevent the user from accidentally overwriting an history
+  const buildImageName = selectedImage.split(':')[0];
   const buildID = await bootcClient.generateUniqueBuildID(buildImageName);
 
   // The build options
+  const image = findImage(selectedImage);
   const buildOptions: BootcBuildInfo = {
     id: buildID,
     image: buildImageName,
-    tag: buildTag,
-    engineId: buildEngineId,
+    tag: selectedImage.split(':')[1],
+    engineId: image?.engineId ?? '',
     folder: buildFolder,
     type: buildType,
     arch: buildArch,
   };
+
   buildInProgress = true;
   try {
     // Do not await.. just start the build.
     // the reason being is that the validation / error logic happens in buildDiskImage
     // in the backend and it will error out there as that is where we can console.log
     // as well as notify the user of the error via showErrorMessage / showInformationMessage, etc.
-    bootcClient.buildImage(buildOptions);
+    bootcClient.buildImage(buildOptions, overwrite || true); // TODO
```

Comment:
remove todo? it's always true?

Reply:
Doh, forget to remove this at the end. Fixed in new commit.